### PR TITLE
Create CircuitPython Community CoC.

### LINKED ISSUE #141: Community library - Code of Conduct
When generating a community library with ~circup~ `cookiecutter` it uses the standard `CODE_OF_CONDUCT.md` that it uses for Adafruit libraries.

Example: https://github.com/lesamouraipourpre/Community_CircuitPython_TCA9555/blob/HEAD/CODE_OF_CONDUCT.md

There are parts in this which seem to be Adafruit specific in particular the *Moderation* and *Scope* sections. While I would obviously respect any reports received through these channels, I feel this puts undue responsibility on the Adafruit Team/Adafruit Discord Team.

Should this use a less branded, library specific Code of Conduct, possibly a 'CircuitPython Community Code of Conduct' (CCCoC) and should ~circup~ `cookiecutter` be amended to provide this version?

If we were to go for a generic CCCoC, what should it say in these sections?

### PR DESCRIPTION
**Pull request by kattni**
Fixes #141.

Please do not merge this until the conversation has been concluded.

I made a few simple changes to get started - swapping out "CircuitPython" for "Adafruit" in a few places, and making some of the wording more generalised. "Adafruit" remains in places where we need to make specific decisions about content.

I'm realising that, unfortunately, since a second file needed to be generated to make `cookiecutter` generate the proper CoC depending on library-type choice, it doesn't render it as a diff from the original.

None of the changes I made need to stay in - I simply wanted to create a starting point for this conversation.

### CODE REVIEW

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:91]:
```suggestion
you should also report the message directly to [Discord](https://discord.com/safety).
```

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:96]:
```suggestion
   CircuitPython Community Code of Conduct. All reports will be reviewed and
```

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:113]:
```suggestion
   the Adafruit GitHub organisation, Adafruit Forums, Adafruit Twitter, as well as other community sites, etc.
```

**Review by askpatrickw — COMMENTED** [review]
I had minor suggestions, but overall @kattni I think it looks great!

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (kattni) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:96]: I think I skipped this one because I expected this whole section to change. But on second read through, this bit is pretty easily generalised so this simple change will make it work.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (deshipu) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:26]: Would it make sense to make this use the same tense as the other points, that is, "Being kind and courteous to others"?

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (jposada202020) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:89]: Could we use a different word for `egregious` as a no native English speaker this is kind of hard.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (kattni) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:26]: Yes it probably would make sense.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (kattni) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:89]: Sure, I'll sort out a simpler word.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:26]:
```suggestion
* Being kind and courteous to others
```

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:113]:
```suggestion
   community platforms. For example, a severe violation in one Community forum
   may result in a ban on not only the CircuitPython GitHub organisation,
   CircuitPython Twitter, but also on the CircuitPython channels on Adafruit's Discord, etc.
```

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:89]: awful, deplorable, blatant, flagrant  ?
https://www.merriam-webster.com/thesaurus/egregious

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:80]: Suggestion: setup a community@circuitpython.org email inbox that just forwards to support@adafruit.com for now.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:98]: egregious is used here as well.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (askpatrickw) [inline {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}:121]:
```suggestion
chats on the CircuitPython channels hosted on Adafruit Discord, or interaction at a conference.
```

**Comment by kattni** [comment]
Hey, folks. I think a clarification of the scope of this Code of Conduct is needed before we continue. I may be the one who is unclear on what was being proposed or what is desired.

It is my understanding that the idea was to create a CoC that applied to libraries added to the Community Bundle. If that is the case, the moderation responsibilities fall to the maintainer of the particular project, not to Adafruit. Therefore continuing to include support(at)adafruit.com or any email that forwards to them would defeat the purpose. As well, referencing the Adafruit Discord also defeats the purpose, as any issue that occurs there is within Adafruit's scope, and within the Code of Conduct listed on Discord. 

Any action taken by a community maintainer on their own project could be applied to the Adafruit community as well (e.g. if the violation is serious enough, Adafruit may reciprocate the resolution), but Adafruit should not be taking action on behalf of the community maintainer. As well, if a violation of the Adafruit CoC somehow relates to a maintainer project (e.g. a discussion on Discord about the project results in a CoC violation), the maintainer can reciprocate Adafruit's action as well.

If the scope is meant to apply to the project and therefore responsibility is meant to fall to the project maintainer, then we need to generalise the wording to, in the event of a violation, suggest filing an issue on a repository, or emailing the project maintainer directly. The Discord verbiage should likely be removed entirely, as, again, any actions on Discord are covered by the Adafruit CoC. I believe there is more that should be changed, but this should at least give you an idea of what my understanding is.

Please let me know what the intention was for this CircuitPython Community Code of Conduct.

**Comment by jposada202020** [comment]
My understanding was to make a more general CoC, to be used for the community Bundle as described in issue #141

**Comment by askpatrickw** [comment]
> My understanding was to make a more general CoC, to be used for the community Bundle as described in issue #141

And I would say for the Community in General... and the places the Community is hosted which would include Adafruit but not _just_ Adafruit. If I was doing a livestream on CP (will never happen) and someone showed up and was offensive, that would mean the same things as doing it on Adafruit Discord.

If I was doing a livestream on making the best peanut butter and jelly sandwich (still won't happen) then it wouldn't have anything to do with CP or Adafruit unless I guess the person involved was the same in both cases.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (kattni) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}]: I have updated this to "the project maintainer." We can discuss this further.

File: {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}
Comment (kattni) [reply {% if cookiecutter.library_prefix %}{{ cookiecutter.library_prefix | capitalize }}_CircuitPython{% else %}CircuitPython_Org{% endif %}_{{ cookiecutter.library_name|replace(" ", "_")}}/{% if cookiecutter.target_bundle == 'Community' %}CODE_OF_CONDUCT.md{% endif %}]: I used "offensive".

**Review by jposada202020 — APPROVED** [review]

**Review by askpatrickw — APPROVED** [review]
I think this looks really good!

**Comment by kattni** [comment]
@lesamouraipourpre Would you also like to take a look? Thanks!

**Review by lesamouraipourpre — APPROVED** [review]
The proposals look good to me. Great work!

**Comment by kattni** [comment]
Thanks for the feedback! I've asked @FoamyGuy to test this to make sure both the Community and Adafruit methods build with the appropriate CoC (I'm not super experienced with Jinja, so I want to make sure I did it properly). Please hold off on merging until that test is completed. Thank you!

**Review by FoamyGuy — APPROVED** [review]
These changes look good to me. 

I successfully tested the 3 scenarios:

-  Adafruit Library
- CircuitPython org library
- 3rd party Individual (Foamyguy) library

and verified the Code of Conduct is showing the correct contents in all of these cases. 

Thanks for working on this @kattni

**Pull request merged**
